src/test/groovy/org/prebid/server/functional/tests/privacy/TcfBasicTransmitEidsAligningActivitiesSpec.groovy

```diff
@@ -0,0 +1,568 @@
+package org.prebid.server.functional.tests.privacy
+
+import org.prebid.server.functional.model.bidder.Generic
+import org.prebid.server.functional.model.config.AccountGdprConfig
+import org.prebid.server.functional.model.config.Purpose
+import org.prebid.server.functional.model.privacy.EnforcementRequirements
+import org.prebid.server.functional.model.request.auction.Activity
+import org.prebid.server.functional.model.request.auction.ActivityRule
+import org.prebid.server.functional.model.request.auction.AllowActivities
+import org.prebid.server.functional.model.request.auction.Condition
+import org.prebid.server.functional.model.request.auction.Eid
+import org.prebid.server.functional.service.PrebidServerService
+import org.prebid.server.functional.util.privacy.TcfUtils
+
+import static org.prebid.server.functional.model.bidder.BidderName.ALIAS
+import static org.prebid.server.functional.model.bidder.BidderName.GENERIC
+import static org.prebid.server.functional.model.config.Purpose.P1
+import static org.prebid.server.functional.model.config.Purpose.P2
+import static org.prebid.server.functional.model.config.Purpose.P3
+import static org.prebid.server.functional.model.config.Purpose.P4
+import static org.prebid.server.functional.model.config.Purpose.P5
+import static org.prebid.server.functional.model.config.Purpose.P6
+import static org.prebid.server.functional.model.config.Purpose.P7
+import static org.prebid.server.functional.model.config.Purpose.P8
+import static org.prebid.server.functional.model.config.Purpose.P9
+import static org.prebid.server.functional.model.config.Purpose.P10
+import static org.prebid.server.functional.model.config.PurposeEnforcement.BASIC
+import static org.prebid.server.functional.model.config.PurposeEnforcement.NO
+import static org.prebid.server.functional.model.request.auction.ActivityType.TRANSMIT_EIDS
+import static org.prebid.server.functional.model.request.auction.TraceLevel.VERBOSE
+import static org.prebid.server.functional.util.privacy.TcfConsent.GENERIC_VENDOR_ID
+
+class TcfBasicTransmitEidsAligningActivitiesSpec extends PrivacyBaseSpec {
+
+    private static final Map<String, String> PBS_CONFIG = SETTING_CONFIG + GENERIC_VENDOR_CONFIG + GENERIC_COOKIE_SYNC_CONFIG + ["gdpr.vendorlist.v2.http-endpoint-template": null,
+                                                                                                                                 "gdpr.vendorlist.v3.http-endpoint-template": null]
+
+    private final PrebidServerService activityPbsServiceExcludeGvl = pbsServiceFactory.getService(PBS_CONFIG)
+
+    private final PrebidServerService activityPbsServiceExcludeGvlWithElderOrtb = pbsServiceFactory.getService(PBS_CONFIG + ["adapters.generic.ortb-version": "2.5"])
```

Comment:
I haven't reviewed the whole PR yet, but I'd leave it be. Because otherwise, we won't be able to tell for sure whether masking happens before or after the normalization. But maybe it would be worth moving those tests to a separate location.